### ISSUE TITLE
Member.voice.setChannel() disconnects

### ISSUE THREAD

**Issue by knyzorg** [discordjs/discord.js#6934]
### Issue description

I'm trying to use member.voice.setChannel() but it disconnects the user instead of moving them

Logs output: 

```
slava joined 904859594474536974
slava joined 904873537762914315
Moving to 904859594474536974
slava left 904873537762914315
```

### Code sample

```typescript
bot.on("voiceStateUpdate", (oldState, newState) => {
  if (newState.channelId) {
    console.log(oldState.member.displayName, "joined", newState.channelId);
    if (oldState.channelId !== null) {
      console.log("Moving to", oldState.channelId);
      newState.member.voice.setChannel(oldState.channel);
    }
  } else {
    console.log(oldState.member.displayName, "left", oldState.channelId);
  }
});
```

### discord.js version

13.3.1

### Node.js version

latest

### Operating system

Linux

### Priority this issue should have

Medium (should be fixed soon)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

GUILD_VOICE_STATES

**Comment by Jiralite** [comment]
The problem here is that discord.js requires the `GUILDS` intent. discord.js will attempt to resolve the channel from the guild, but it will be `null` because of the lack of the guild. Providing the channel's id won't work either, as it'll internally later be attempted to be resolved as a `BaseGuildVoiceChannel`.

TL;DR: use the `GUILDS` intent and your problem goes away.
